### ISSUE TITLE
Getting Started with Neural Networks

### ISSUE THREAD

**Issue by arthurW1935** [stackup-dev/open-source-campaign#120]
# Name of Proposer
arthur

# Campaign Name
Getting Started with Neural Networks

## Campaign Description
- This campaign aims to teach the stackies the basics of neural networks, and apply that knowledge to create basic model with neural networks.
- This campaign is for everyone who wants to learn :D 
- The campaign will be using Python as the code language
- Stackies should join this quest as neural networks are one of the most important and widely used technology in machine learning.

#### This campaign should have 4 quests
- Quest 1: Introduction to Neural Networks and Tensorflow
- Quest 2: Solving an equation 
- Quest 2: Introduction to ANN
- Quest 3: The Fashion-MNIST Dataset

## Quest 1 Summary
- This quest will be focusing on introducing neural networks, and the widely used library Tensorflow and Keras
- Resources to be used:
Neural Networks: https://www.section.io/engineering-education/introduction-to-neural-networks/
Tensorflow: https://www.javatpoint.com/tensorflow-introduction
- This quest will be the basic conception that a stackie needs to have for the upcoming quests

## Quest 1 Learning Objectives
By the end of the quest, a stackie will know
- basics of neural networks
- what TensorFlow is
- the importance of Tensorflow

## Quest 1 Deliverables
A quiz consisting of 10 - 12 questions.

## Quest 2 Summary
- This quest will be focusing on using TensorFlow Keras sequential model to train a model a linear equation
- Code:
```
import tensorflow as tf
import numpy as np
from tensorflow import keras
```
```
model = tf.keras.Sequential([keras.layers.Dense(units=1, input_shape=[1])])
model.compile(optimizer='sgd', loss='mean_squared_error')
```
```
x = np.array([-1.0, 0.0, 1.0, 2.0, 3.0, 4.0], dtype=float)
y = np.array([-2.00, 1.0, 2.0, 3.0, 4.0], dtype=float)
y = np.array([-2.0, 1.0, 4.0, 7.0, 10.0, 13.0], dtype=float)
model.fit(x, y, epochs=500)
```
```
print(model.predict([10.0]))
```
- This quest will be the basic implementation that will clear the concepts of a stackie and also will be interesting.

## Quest 2 Learning Objectives
By the end of the quest, a stackie will know
- what Keras Sequential is
- what Dense layers are
- SGD optimizer and Mean Squared Error loss function
- how to train and test the model

## Quest 2 Deliverables
A screenshot of the code window and the predicted output.

## Quest 3 Summary
- This quest will be focusing on explaining how artificial neural networks work
- Resources:
ANN : https://www.datasciencecentral.com/artificial-neural-network-ann-in-machine-learning/
- This quest will be the basic conception that a stackie needs to have for the upcoming quests

## Quest 3 Learning Objectives
By the end of the quest, a stackie will know
- what artificial neural networks are
- input, hidden and output layer
- its advantages and disadvantages

## Quest 3 Deliverables
A quiz consisting of 10 - 12 questions

## Quest 4 Summary
- This quest will be focusing on working with Fashon-MNIST dataset using TensorFlow Keras 
- Code:
```
import tensorflow as tf
mnist = tf.keras.datasets.fashion_mnist
(training_images, training_labels), (test_images, test_labels) = mnist.load_data()
```
```
import matplotlib.pyplot as plt
plt.imshow(training_images[0])
print(training_labels[0])
print(training_images[0])
```
```
training_images  = training_images / 255.0
test_images = test_images / 255.0
```
```
model = tf.keras.models.Sequential([tf.keras.layers.Flatten(), 
                                    tf.keras.layers.Dense(128, activation=tf.nn.relu), 
                                    tf.keras.layers.Dense(10, activation=tf.nn.softmax)])

model.compile(optimizer = tf.keras.optimizers.Adam(),
              loss = 'sparse_categorical_crossentropy',
              metrics=['accuracy'])

model.fit(training_images, training_labels, epochs=5)
```
```
model.evaluate(test_images, test_labels)
```
- This quest will be the final capstone project of the campaign

## Quest 4 Learning Objectives
By the end of the quest, a stackie will know
- the Fashion-MNIST dataset
- training and test data
- relu and softmax activation functions
- Adam optimiser and Sparse Categorical Crossentropy loss function
- evaluate method for testing

## Quest 4 Deliverables
A screenshot of the code window and the predicted output.
